libs/langchain/langchain/document_loaders/parsers/pdf.py

```diff
@@ -263,24 +263,94 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
 
 class DocumentIntelligenceParser(BaseBlobParser):
     """Loads a PDF with Azure Document Intelligence
-    (formerly Forms Recognizer) and chunks at character level."""
+    (formerly Forms Recognizer). Returns Document with
+    pages or paragraphs, table headers, and rows."""
 
-    def __init__(self, client: Any, model: str):
+    def __init__(self, client: Any, model: str, split_mode: str):
         self.client = client
         self.model = model
+        self.split_mode = split_mode
 
     def _generate_docs(self, blob: Blob, result: Any) -> Iterator[Document]:
-        for p in result.pages:
-            content = " ".join([line.content for line in p.lines])
-
-            d = Document(
-                page_content=content,
-                metadata={
-                    "source": blob.source,
-                    "page": p.page_number,
-                },
-            )
-            yield d
+        page_content_dict = dict()
+
+        for paragraph in result.paragraphs:
+            page_number = paragraph.bounding_regions[0].page_number
+
+            if self.split_mode == "page":
+                if page_number not in page_content_dict:
+                    page_content_dict[page_number] = str()
+
+                page_content_dict[page_number] += paragraph.content + "\n\n"
+            elif self.split_mode == "paragraph":
+                d = Document(
+                    page_content=paragraph.content,
+                    metadata={
+                        "source": blob.source,
+                        "page": page_number,
+                        "type": "PARAGRAPH",
+                    },
+                )
+                yield d
+
+        if self.split_mode == "page":
+            for page, content in page_content_dict.items():
+                d = Document(
+                    page_content=content.strip(),
+                    metadata={
+                        "source": blob.source,
+                        "page": page,
+                        "type": "PAGE",
+                    },
+                )
+                yield d
```

Comment:
@baskaryan here's `page` vs `paragraph` logic. If `page` is used then we do subsequent collation of paragraphs into individual page's full text and specify `"type": "PAGE"` in `Document`. If `paragraph` is used then we keep it as is and simply `yield` with paragraphs in the `Document` schema with `"type": "PARAGRAPH"`